### PR TITLE
Support CvmFS 2.11.0 parameters

### PR DESCRIPTION
**Pull request by traylenator**
#### Pull Request (PR) description

With the release of CvmFS 2.11.0 there are 8 new client
parameters that can be set:

For instance:

```puppet
class{'cvmfs':
  cvmfs_cache_symlinks => 'yes',
  cvmfs_streaming_cache => 'no',
  cvmfs_statfs_cache_timeout => 10,
  cvmfs_world_readable => 'yes',
  cvmfs_cpu_affinity => [0,1,2],
  cvmfs_xattr_privileged_gids => [100,101,102],
  cvmfs_xattr_protected_xattrs => ['user.foo','user.bar'],
  cvmfs_cache_refcount => 'yes',
}
```

* CvmFS 2.11.0 release notes https://cvmfs.readthedocs.io/en/stable/cpt-releasenotes.html
* CvmFS parameters https://cvmfs.readthedocs.io/en/stable/apx-parameters.html

Couple of extra commits fix some typos and lint correction on parameter order.

### CODE REVIEW

**Comment by jblomer** [comment]
This is missing the `CVMFS_CACHE_REFCOUNT` parameter, isn't it?

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -351,6 +351,51 @@
             end
           end
 
+          context 'with serveral parameters not set' do
```
Comment (olifre) [inline spec/classes/init_spec.rb:354]: (harmless) typo: `serveral` => `several`

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -351,6 +351,51 @@
...
+                without_content(%r{CVMFS_STREAMING_CACHE}).
+                without_content(%r{CVMFS_STATFS_CACHE_TIMEOUT}).
+                without_content(%r{CVMFS_WORLD_READABLE}).
+                without_content(%r{CVMFS_CPU_AFFINITY}).
+                without_content(%r{CVMFS_XATTR_PRIVILEGED_GIDS$}).
+                without_content(%r{CVMFS_XATTR_PRIVILEGED_XATTRS$})
+            end
+          end
+
+          context 'with serveral parameters set' do
```
Comment (olifre) [inline spec/classes/init_spec.rb:373]: Same here :wink:.

File: spec/classes/init_spec.rb [hunk 3]
```diff
@@ -351,6 +351,51 @@
...
+
+            it do
+              is_expected.to contain_concat__fragment('cvmfs_default_local_header').
+                without_content(%r{CVMFS_CACHE_SYMLINKS}).
+                without_content(%r{CVMFS_STREAMING_CACHE}).
+                without_content(%r{CVMFS_STATFS_CACHE_TIMEOUT}).
+                without_content(%r{CVMFS_WORLD_READABLE}).
+                without_content(%r{CVMFS_CPU_AFFINITY}).
+                without_content(%r{CVMFS_XATTR_PRIVILEGED_GIDS$}).
+                without_content(%r{CVMFS_XATTR_PRIVILEGED_XATTRS$})
```
Comment (olifre) [inline spec/classes/init_spec.rb:369]: Are the traling `$` needed here?

**Review by olifre — COMMENTED** [review]
That looks good to me in general, just a few nits in the individual comments. Anf of course I second @jblomer 's question, it would be nice to add `CVMFS_CACHE_REFCOUNT` while at it :+1: .

**Comment by traylenator** [comment]
> CV

**Comment by traylenator** [comment]
CVMFS_CACHE_REFCOUNT added.  ( closed by accident)

**Review by olifre — APPROVED** [review]
Thanks, that looks good to me :+1: .

**Pull request merged**